samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/evaluation/TotalScoresBean.java

```diff
@@ -265,38 +269,41 @@ public boolean getIsOneSelectionType() {
    * @return results - Map
    */
   public Map getResults() {
-    // Instance a new PublishedAssessmentService to get all the published Answer for each student
-    PublishedAssessmentService pubAssessmentService = new PublishedAssessmentService();
-    Map publishedAnswerHash = pubAssessmentService.preparePublishedAnswerHash(pubAssessmentService.getPublishedAssessment(this.getPublishedId()));
-    // Instance a new GradingService to get all the student responses
-    GradingService gradingService = new GradingService();
-    Map<Long, List<Integer>> resultsByUser = new HashMap<>();
-    // For each agent (student) we will search the correct/incorrect/empty responses
-    for (Object object : agents) {
-      AgentResults agentResults = (AgentResults) object;
-      if (agentResults.getAssessmentGradingId() != -1) {
-        // Getting the responses for that student (agentResults)
-        AssessmentGradingData assessmentGradingAux = gradingService.load(agentResults.getAssessmentGradingId().toString());
-        List<Integer> resultsAux = new ArrayList<>(Collections.nCopies(3, 0));
-        for (ItemGradingData item : assessmentGradingAux.getItemGradingSet()) {
-          if (item.getPublishedAnswerId() == null) { // If it does not have publishedAnswerId that means it is empty
-            resultsAux.set(2, resultsAux.get(2) + 1);
-          } else { // If it has publishedAnswerId that means has response
-            // If it has response we will get the answer from publishedAnswerHash and if it is correct or incorrect
-            AnswerIfc answer = (AnswerIfc) publishedAnswerHash.get(item.getPublishedAnswerId());
-            if (!answer.getIsCorrect()) {
-              // For incorrect answers cases
-              resultsAux.set(1, ((int) resultsAux.get(1)) + 1);
-            } else {
-              // For correct answers cases
-              resultsAux.set(0, ((int) resultsAux.get(0)) + 1);
+    if (!this.isEnteredInResults()) {
+      this.setEnteredInResults(true);
```

Comment:
What does this mean? Like it executed this portion before? Maybe something like this.resultsAlreadyCalculated() ?

Reply:
Yes, it's like you say. But I prefer your variable's name. I will change it